mockgcp/mockkms/keyhandle.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +mockgcp-support
+// apiVersion: kms.cnrm.cloud.google.com/v1beta1
+// kind: KMSAutokeyConfig
+// service: google.cloud.kms.v1.AutokeyAdmin
+// resource: AutokeyConfig
+
+package mockkms
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	lro "cloud.google.com/go/longrunning/autogen/longrunningpb"
+	"google.golang.org/protobuf/proto"
+
+	pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/cloud/kms/v1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/pkg/storage"
+)
+
+type autokeyServer struct {
+	*MockService
+	pb.UnimplementedAutokeyServer
+}
+
+func (r *autokeyServer) GetKeyHandle(ctx context.Context, req *pb.GetKeyHandleRequest) (*pb.KeyHandle, error) {
+	parent, resourceID, err := r.parseKeyHandleName(req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := parent.String() + "/keyHandles/" + resourceID
+
+	obj := &pb.KeyHandle{}
+	if err := r.storage.Get(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	return obj, nil
+}
+
+func (r *autokeyServer) CreateKeyHandle(ctx context.Context, req *pb.CreateKeyHandleRequest) (*lro.Operation, error) {
+	parent, resourceID, err := r.parseKeyHandleName(req.KeyHandle.GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := parent.String() + "/keyHandles/" + resourceID
+
+	obj := proto.Clone(req.GetKeyHandle()).(*pb.KeyHandle)
+	obj.Name = fqn
+
+	if err := r.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+	return r.operations.StartLRO(ctx, parent.String(), nil, func() (proto.Message, error) {
+		result := proto.Clone(obj).(*pb.KeyHandle)
+		return result, nil
+	})
+}
+
+func (r *autokeyServer) ListKeyHandles(ctx context.Context, req *pb.ListKeyHandlesRequest) (*pb.ListKeyHandlesResponse, error) {
```

Comment:
Do you mean there is not DELETE and UPDATE methods for this resource?

Reply:
yes